tests/core/test_indexset.py

```diff
@@ -106,11 +105,17 @@ def test_list_indexsets(self, test_mp, request):
         ]
         assert not (set(expected_id) ^ set(list_id))
 
+        # Test that only indexsets belonging to this Run are listed
```

Comment:
Wouldn't it be much less verbose to add the following after creating `indexset_2`?

```python
# create an indexset in another run to make sure that the correct indexsets are listed
test_mp.runs.create("Model", "Other Scenario").optimization.indexsets.create("Other Indexset")
```

Reply:
Very good catch, I'll rework the tests here and in the other PRs.